File: service/internal/models/uploads.go
package models

import "time"

type UploadRecord struct {
	ID          int64     `json:"id,omitempty"`
	SenderID    string    `json:"sender_id"`
	SenderIP    string    `json:"sender_ip"`
	Filename    string    `json:"filename"`
	Filesize    int64     `json:"filesize"`
	SendTime    time.Time `json:"send_time"`
	ReceiverID  string    `json:"receiver_id"`
	ReceiverIP  string    `json:"receiver_ip"`
	ReceiveTime time.Time `json:"receive_time"`
}

File: web/src/types/uploads.ts
export interface UploadRecord {
    id?: number;
    sender_id: string;
    sender_ip: string;
    filename: string;
    filesize: number;
    send_time: string;
    receiver_id: string;
    receiver_ip: string;
    receive_time: string;
  }
  
  export interface PaginatedUploadRecords {
    records: UploadRecord[];
    total_count: number;
    total_pages: number;
    page: number;
    page_size: number;
  }